tools/prom2influx.go

```diff
@@ -0,0 +1,154 @@
+// This tool receives JSON metrics of Prometheus from stdin and writes them
+// to a influxdb server
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"time"
+
+	influx "github.com/influxdata/influxdb/client/v2"
+)
+
+type options struct {
+	Host   string
+	Port   string
+	User   string
+	Passwd string
+	DBName string
+	Chunk  int
+}
+
+func parseOpts() options {
+	influxHost := flag.String("host", "localhost", "The host of influxdb.")
+	influxPort := flag.String("port", "8086", "The port of influxdb.")
+	influxUser := flag.String("user", "", "The username of influxdb.")
+	influxPass := flag.String("passwd", "", "The password of user.")
+	influxDB := flag.String("db", "tidb-insight", "The database name of imported metrics.")
+	influxChunk := flag.Int("chunk", 2000, "The chunk size of writing.")
+	flag.Parse()
+
+	var opts options
+	opts.Host = *influxHost
+	opts.Port = *influxPort
+	opts.User = *influxUser
+	opts.Passwd = *influxPass
+	opts.DBName = *influxDB
+	opts.Chunk = *influxChunk
+	return opts
+}
+
+// queryDB convenience function to query the database
+func queryDB(clnt influx.Client, db_name string, cmd string) (res []influx.Result, err error) {
+	q := influx.Query{
+		Command:  cmd,
+		Database: db_name,
+	}
+	if response, err := clnt.Query(q); err == nil {
+		if response.Error() != nil {
+			return res, response.Error()
+		}
+		res = response.Results
+	} else {
+		return res, err
+	}
+	return res, nil
+}
+
+func slicePoints(data []map[string]interface{}, chunkSize int) [][]map[string]interface{} {
```

Comment:
It seems that metrics of promeths are grouped by instance, why need 2000 chunkSize?